src/common/actions/repository-input.js

```diff
@@ -52,3 +56,49 @@ export function verifyGitHubRepositoryError(error) {
     error: true
   };
 }
+
+export function createSnap(repository, location) {
+  return (dispatch) => {
+    if (repository) {
+      dispatch({
+        type: CREATE_SNAP,
+        payload: repository
+      });
+
+      const BASE_URL = conf.get('BASE_URL');
+      return fetch(`${BASE_URL}/api/launchpad/snaps`, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({
+          repository_url: `https://github.com/${repository}`
+        }),
+        credentials: 'same-origin'
+      })
+        .then(checkStatus)
+        .then(response => {
+          return response.json().then(result => {
+            if (result.status !== 'success' ||
+                result.payload.code !== 'snap-created') {
+              const error = new Error(response.statusText);
+              error.response = response;
+              throw error;
+            }
+            const startingUrl = `${BASE_URL}/${repository}/builds`;
+            (location || window.location).href =
+              `${BASE_URL}/login/authenticate` +
+              `?starting_url=${encodeURIComponent(startingUrl)}` +
+              `&caveat_id=${encodeURIComponent(result.payload.message)}`;
```

Comment:
Location can be changed using `react-router-redux`, which loads a different route without a full page request. There's an example of this in my PR [here](https://github.com/canonical-ols/build.snapcraft.io/pull/41/files#diff-4643f606293beb5e4781b5aa11c170e0R30).

This requires the router middleware to be added to the store, which has not yet been merged into master. An example of this can be found [here](https://github.com/canonical-ols/build.snapcraft.io/pull/41/files#diff-6a5ca9964ee5cf49b3b87af20da794e4).